### PR TITLE
Fix internal-debug syncz not working with agent, and add support for debug type `config_dump `

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Currently if I do:
```
istioctl x internal-debug syncz  --xds-address localhost:15004 --plaintext
```
I get no response.

Also, replace `x ps` with `x internal-debug syncz`. This is a prerequisite of https://github.com/istio/istio/pull/47740.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [x] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [x] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Review by hzxuzhonghu — COMMENTED** [review]
I am not even aware there is an internal-debug sub command, do we have any docs related

**Comment by hanxiaop** [comment]
> I am not even aware there is an internal-debug sub command, do we have any docs related

@hzxuzhonghu I don't think so. It is a command marked as hidden. I haven't investigated the intention of the command further, but right now the command example seems to use only `syncz` to provide debug interface via xds, which is identical to `istioctl x ps`.

**Comment by hzxuzhonghu** [comment]
Is it just similar? Can we use istioctl x ps to replace that one

**Comment by hanxiaop** [comment]
> Is it just similar? Can we use istioctl x ps to replace that one

@hzxuzhonghu Sorry for the misunderstanding, I meant that `istioctl x internal-debug syncz` is identical to `istioct x ps`, `x ps` is like a subset of `internal-debug`. The internal-debug command can be used to obtain additional info like `config_dump`, but that will require to use resource name which will be like `istioctl x internal-debug config_dump <resource name>`.

**Comment by hanxiaop** [comment]
@hzxuzhonghu Basically it's generate through:

https://github.com/istio/istio/blob/572acf2859888645467345cb4c7aa418176b757d/pilot/pkg/xds/statusgen.go#L83-L85

File: tests/integration/pilot/piggyback_test.go [hunk 1]
```diff
@@ -86,7 +86,13 @@ func TestPiggyback(t *testing.T) {
 				defer pf.Close()
 
 				istioCtl := istioctl.NewOrFail(t, t, istioctl.Config{Cluster: t.Clusters().Default()})
-				args := []string{"x", "proxy-status", "--plaintext", "--xds-address", pf.Address()}
+				args := []string{
```
Comment (GregHanson) [inline tests/integration/pilot/piggyback_test.go:89]: thanks for the fix

**Review by GregHanson — APPROVED** [review]
was also unaware of this internal-debug command

**Pull request merged**